src/data/finset/lattice.lean

```diff
@@ -429,7 +432,49 @@ lemma inf_sup_distrib_right (s : finset ι) (f : ι → α) (a : α) :
   s.inf f ⊔ a = s.inf (λ i, f i ⊔ a) :=
 @sup_inf_distrib_right αᵒᵈ _ _ _ _ _ _
 
+protected lemma codisjoint_inf_right : codisjoint a (s.inf f) ↔ ∀ ⦃i⦄, i ∈ s → codisjoint a (f i) :=
+by simp only [codisjoint_iff, inf_sup_distrib_left, finset.inf_eq_top_iff]
+
+protected lemma codisjoint_inf_left : codisjoint (s.inf f) a ↔ ∀ ⦃i⦄, i ∈ s → codisjoint (f i) a :=
+by simp only [codisjoint_iff, inf_sup_distrib_right, finset.inf_eq_top_iff]
+
+lemma inf_sup_inf (s : finset ι) (t : finset κ) (f : ι → α) (g : κ → α) :
+  s.inf f ⊔ t.inf g = (s ×ˢ t).inf (λ i, f i.1 ⊔ g i.2) :=
+by simp_rw [finset.inf_sup_distrib_right, finset.inf_sup_distrib_left, inf_product_left]
```

Comment:
I don't have gitpod access right now, but I want to check how painful an order_dual golf is here.

Reply:
Oh it works, but I prefer to avoid using `order_dual` when the proof is already a one-liner.